storage/remote/write_handler.go

```diff
@@ -324,16 +324,85 @@ const (
 	rw20WrittenExemplarsHeader  = "X-Prometheus-Remote-Write-Written-Exemplars"
 )
 
-type responseStats struct {
-	samples    int
-	histograms int
-	exemplars  int
+// WriteResponseStats represents the response write statistics specified in https://github.com/prometheus/docs/pull/2486
+type WriteResponseStats struct {
+	// Samples represents X-Prometheus-Remote-Write-Written-Samples
+	Samples int
+	// Histograms represents X-Prometheus-Remote-Write-Written-Histograms
+	Histograms int
+	// Exemplars represents X-Prometheus-Remote-Write-Written-Exemplars
+	Exemplars int
+
+	// Confirmed means we can trust those statistics from the point of view
+	// of the PRW 2.0 spec. When parsed from headers, it means we got at least one
+	// response header from the Receiver to confirm those numbers, meaning it must
+	// be a at least 2.0 Receiver. See ParseWriteResponseStats for details.
+	Confirmed bool
 }
 
-func (s responseStats) SetResponseHeaders(h http.Header) {
-	h.Set(prw20WrittenSamplesHeader, strconv.Itoa(s.samples))
-	h.Set(rw20WrittenHistogramsHeader, strconv.Itoa(s.histograms))
-	h.Set(rw20WrittenExemplarsHeader, strconv.Itoa(s.exemplars))
+// NoDataWritten returns true if statistics indicate no data was written.
+func (s WriteResponseStats) NoDataWritten() bool {
+	return (s.Samples + s.Histograms + s.Exemplars) == 0
+}
+
+// AllSamples returns both float and histogram sample numbers.
+func (s WriteResponseStats) AllSamples() int {
+	return s.Samples + s.Histograms
+}
+
+// Add returns the sum of this WriteResponseStats plus the given WriteResponseStats.
+func (s WriteResponseStats) Add(rs WriteResponseStats) WriteResponseStats {
+	s.Confirmed = rs.Confirmed
```

Comment:
nit: I'm not sure this looks right. Should we maybe set it to false when we add at least one that is false? Or maybe if that's not possible to add a false one, be clear and always set to true?

Reply:
Yea, this is tricky. I am not sure how retries should work with those stats, it depends if on the receiver independence etc. Shouldn't be a big deal too, we might never got it right, but some consistency would be nice.

My thinking is that, if it was once confirmed it's likely it's generally 2.0 Receiver, so all is confirmed? Maybe `confirm` is a too generic name and we should actually name it `likely2.0` or something

Reply:
@TheSpiritXIII not sure I understand your comment, add one of _what_ that is false?